### PR TITLE
x11-libs/fltk: add 1.3.8

### PR DESCRIPTION
**Pull request by vowstar**
bump fltk to 1.3.8

Closes: https://bugs.gentoo.org/847607

### CODE REVIEW

File: x11-libs/fltk/fltk-1.3.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+		$(use_enable opengl gl) \
+		$(use_enable threads) \
+		$(use_enable xft) \
+		$(use_enable xinerama) \
+		--disable-localjpeg \
+		--disable-localpng \
+		--disable-localzlib \
+		--docdir="${EPREFIX}/usr/share/doc/${PF}/html" \
+		--enable-largefile \
+		--enable-shared \
```
Comment (juippis) [inline x11-libs/fltk/fltk-1.3.8.ebuild:122]: This makes the `static-libs` use flag seem a bit redundant.

File: x11-libs/fltk/fltk-1.3.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+		docinto examples
+		dodoc -r test/*.{h,cxx,fl} test/demo.menu
+	fi
+
+	insinto /usr/share/cmake/Modules
+	doins CMake/FLTK*.cmake
+
+	echo "LDPATH=${FLTK_LIBDIRS}" > 99fltk || die
+	echo "FLTK_DOCDIR=${EPREFIX}/usr/share/doc/${PF}/html" >> 99fltk || die
+	doenvd 99fltk
```
Comment (juippis) [inline x11-libs/fltk/fltk-1.3.8.ebuild:181]:
```suggestion
	newenvd - 99fltk <<- _EOF_
		LDPATH="${FLTK_LIBDIRS}"
		FLTK_DOCDIR="${EPREFIX}"/usr/share/doc/"${PF}"/html
	_EOF_
```

File: x11-libs/fltk/fltk-1.3.8.ebuild [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+pkg_postinst() {
+	xdg_desktop_database_update
+	xdg_icon_cache_update
+}
+
+pkg_postrm() {
+	xdg_desktop_database_update
+	xdg_icon_cache_update
+}
```
Comment (juippis) [inline x11-libs/fltk/fltk-1.3.8.ebuild:200]: You can just inherit *xdg.eclass* and remove these lines if you wish.

File: x11-libs/fltk/fltk-1.3.8.ebuild (hunk 2 above)
Comment (vowstar) [reply x11-libs/fltk/fltk-1.3.8.ebuild]: Done, thanks!

File: x11-libs/fltk/fltk-1.3.8.ebuild (hunk 3 above)
Comment (vowstar) [reply x11-libs/fltk/fltk-1.3.8.ebuild]: Done, thanks!

File: x11-libs/fltk/fltk-1.3.8.ebuild (hunk 1 above)
Comment (juippis) [reply x11-libs/fltk/fltk-1.3.8.ebuild:122]: Heh, of course it is :) 

I'll add some comment to the ebuild, thanks for explaining!

**Review by juippis — APPROVED** [review]

File: x11-libs/fltk/fltk-1.3.8.ebuild (hunk 1 above)
Comment (juippis) [reply x11-libs/fltk/fltk-1.3.8.ebuild:122]: Although I don't see a dependency for static-libs on either octave or tigervnc. If nothing **needs** the static libs enabled on this package, maybe we can unconditionally always just remove them?

**Comment by juippis** [comment]
Also please sync 1.4.9999 live package or remove it if it doesn't serve any purpose anymore.

File: x11-libs/fltk/fltk-1.3.8.ebuild (hunk 1 above)
Comment (vowstar) [reply x11-libs/fltk/fltk-1.3.8.ebuild:122]: Sorry, I just misunderstood.

I didn't find any package that really depends on this ``static-lib`` useflag in ::gentoo. Only executables in ``test/``(from ``x11-libs/fltk`` test stage) are linking statically against libfltk.a. Therefore, we need to consider how to safely remove this  ``static-lib``  useflag. I may need to change the link method in ``test/``, or consider using ``test`` useflag to control it.

**Comment by vowstar** [comment]
> Also please sync 1.4.9999 live package or remove it if it doesn't serve any purpose anymore.

Yes, I currently don’t know the intention of the previous maintainer of this live package, and I haven’t found any packages that must use this live package.

**Pull request closed without merging**